step-functions-data-science-sdk/automate_model_retraining_workflow/automate_model_retraining_workflow.ipynb

```diff
@@ -0,0 +1,983 @@
+    "9. Search for your newly created `S3BucketAccessPolicy` policy and select the check box next to it.\n",
+    "10. Choose **Attach policy**. You will then be redirected to the details page for the role.\n",
+    "11. Copy the **Role ARN** at the top of the Summary tab."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# paste the AWS-Glue-S3-Bucket-Access role ARN from above\n",
+    "glue_role = ''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create a Lambda IAM Role\n",
+    "You also need to create an IAM role so that you can create and execute an AWS Lambda function stored in Amazon S3.\n",
+    "\n",
+    "1. Go to the [IAM console](https://console.aws.amazon.com/iam/).\n",
+    "2. Select **Roles** and then **Create role**.\n",
+    "3. Under **Choose the service that will use this role** select **Lambda**.\n",
+    "4. Choose **Next** until you can enter a **Role name**.\n",
+    "5. Enter a name such as `query_training_status-role` and then select **Create role**.\n",
+    "\n",
+    "Next, attach policies to the role you created. The following steps attach policies that provides Lambda access to S3 and read-only access to SageMaker.\n",
+    "\n",
+    "1. Under the **Permissions** tab, click **Attach Policies**.\n",
+    "2. In the search box, type **SageMaker** and select **AmazonSageMakerReadOnly** from the populated list.\n",
+    "3. In the search box type **AWSLambda** and select **AWSLambdaBasicExecutionRole** from the populated list.\n",
+    "4. Choose **Attach policy**. You will then be redirected to the details page for the role.\n",
+    "5. Copy the **Role ARN** at the top of the **Summary**.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# paste the query_training_status-role role ARN from above\n",
+    "lambda_role = ''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare the Dataset\n",
+    "This notebook uses the XGBoost algorithm to automate the classification of unhappy customers for telecommunication service providers. The goal is to identify customers who may cancel their service soon so that you can entice them to stay. This is known as customer churn prediction.\n",
+    "\n",
+    "The dataset we use is publicly available and was mentioned in the book [Discovering Knowledge in Data](https://www.amazon.com/dp/0470908742/) by Daniel T. Larose. It is attributed by the author to the University of California Irvine Repository of Machine Learning Datasets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "project_name = 'ml_deploy'\n",
+    "\n",
+    "data_source = sagemaker.s3.S3Uploader.upload(local_path='./data/customer-churn.csv',\n",
```

Comment:
[Minor] Consider importing this so it's cleaner to write:
```
from sagemaker.s3 import S3Uploader
data_source = S3Uploader.upload([...])
```

Reply:
Good idea. Fixed in the most recent commit,.